server/src/main/java/org/opensearch/common/blobstore/BlobContainer.java

```diff
@@ -124,6 +128,29 @@ default long readBlobPreferredLength() {
      */
     void writeBlob(String blobName, InputStream inputStream, long blobSize, boolean failIfAlreadyExists) throws IOException;
 
+    /**
+     * Used to check whether vendor plugin support for parallel upload of multiple streams is enabled or not.
+     * Returns false by default
+     *
+     * @return If multi-stream parallel uploads are supported
+     */
+    default boolean isMultiStreamUploadSupported() {
```

Comment:
Is it possible to not have this method, but rather `writeBlobByStreams` use existing `writeBlob`. We can still have serial stream writes, right? Every plugin extender can choose to override this method with advanced implementation, but default would still work.

Right now all existing methods involves dealing with operations with remote store. This would become an outlier.

Reply:
That will complicate things. If we remove this method, then core cannot know whether multi-stream is supported. Meaning, core will always send multiple streams and to handle that plugin has to do multiple uploads of streams in serial pointing to same file. Additionally, plugin will have to write complex code to correctly determine the end offset as now it won't the end of a stream but the end of the last stream.

Reply:
Is it possible for the plugin implementer to get this wrong/inconsistent by incorrectly setting the boolean whilst still unable to support.

Reply:
>Additionally, plugin will have to write complex code to correctly determine the end offset as now it won't be the end of a stream but the end of the last stream

Can we make the contract more explicit to ensure if the API is implemented it would need to support multi-stream. Any information that is needed should be a part of the contract